### ISSUE TITLE
`no-unnecessary-polyfills` false positive for `regexp.escape`

### ISSUE THREAD

**Issue by kevinoid** [sindresorhus/eslint-plugin-unicorn#2584]
The [`unicorn/no-unnecessary-polyfills`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-unnecessary-polyfills.md) rule is triggered by importing [regexp.escape](https://www.npmjs.com/package/regexp.escape) in a package with `engines.node: ">=18"`.  I believe this is a false-positive since [RegExp.escape()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/escape) is not supported by any version of Node.js (also confirmed by the [core-js compat table](https://zloirock.github.io/core-js/compat/)).

Full reproduction:

```sh
npm init -y
npm pkg set type=module
npm pkg set engines.node=">=18"
npm install -D eslint eslint-plugin-unicorn
echo "import eslintPluginUnicorn from 'eslint-plugin-unicorn';
export default [eslintPluginUnicorn.configs.recommended];" >eslint.config.js
npm install regexp.escape
echo "import regexpEscape from 'regexp.escape';" >index.js
./node_modules/.bin/eslint .
```

I've bisected the regression to zloirock/core-js@7321cc180c36c1ddde8bb6d89e89c861cc6e5a91 (`core-js-compat@3.41.0`) which added the feature `es.regexp.escape`.

The crux of the issue appears to be that `coreJsCompat.entries['core-js/full/regexp/escape']` is `['es.regexp.escape', 'esnext.regexp.escape']` while `coreJsCompat({ node: '>=18' }).list` contains `'es.regexp.escape'` but not `'esnext.regexp.escape'`, causing `checkFeatures()` to return `true` at https://github.com/sindresorhus/eslint-plugin-unicorn/blob/e48a6203736e56cd2ff752613ed7ce8b5d4f008a/rules/no-unnecessary-polyfills.js#L133

Thanks,
Kevin
